apps/amo/helpers.py

```diff
@@ -237,18 +237,27 @@ def page_name(app=None):
         return _('Add-ons')
 
 
-@register.function
-@jinja2.contextfunction
-def login_link(context):
-    next = context['request'].path
+def link_with_to_param(context, base):
+    next_path = context['request'].path
 
     qs = context['request'].GET.urlencode()
 
     if qs:
-        next += '?' + qs
+        next_path += '?' + qs
```

Comment:
Should we capture hashes too? I guess we don't really use them but someone might in the future.